modules/dreamview/frontend/src/renderer/traffic_control.js

```diff
@@ -0,0 +1,270 @@
+import * as THREE from "three";
+
+import { loadObject } from "utils/models";
+import trafficLightMaterial from "assets/models/traffic_light.mtl";
+import trafficLightObject from "assets/models/traffic_light.obj";
+import stopSignMaterial from "assets/models/stop_sign.mtl";
+import stopSignObject from "assets/models/stop_sign.obj";
+
+const TRAFFIC_LIGHT_SCALE = 0.009;
+const trafficLightScales = {
+    x: TRAFFIC_LIGHT_SCALE,
+    y: TRAFFIC_LIGHT_SCALE,
+    z: TRAFFIC_LIGHT_SCALE
+};
+
+const STOP_SIGN_SCALE = 0.01;
+const stopSignScales = {
+    x: STOP_SIGN_SCALE,
+    y: STOP_SIGN_SCALE,
+    z: STOP_SIGN_SCALE
+};
+
+const EPSILON = 1e-9;
+const TRAFFIC_LIGHT = 'signal';
+const STOP_SIGN = 'stopSign';
+
+export default class TrafficControl {
+    constructor() {
+        this.hash = -1;
+        this.trafficControl = {
```

Comment:
It's a little overgeneralized to call everything TrafficControl (this includes the class name, class data member and function name). Why do we want to process & store trafficLight & stopSign together? There seems to be enough discrepancies that we need to have some confusing and long switch cases to handle them separately later.

Reply:
Traffic light and stop sign have similar processing flow, now I separate their different configs and pass type as function parameters to get relative configs. Hope that looks much cleaner.